### PR TITLE
Upgrade autopublish scripts

### LINKED ISSUE #6: PROPOSAL: publish on NPM from TRAVIS CI
It is not a difficult task, to set up an autopublish:

1. Build package on Travis CI
  * build
  * lint
  * test
2. If it is a **tag** - **deploy package to NPM** (only if there are no errors on test- and lint- steps). Version for the new NPM package will be taken from name of the tag

### PR DESCRIPTION
**Pull request by maxkoryukov**
Added:

* `.npmignore` to skip extra files on publishing (I don't need even `README.md`, because after installation it will be _somewhere_ in `./node_modules/node-freshde...`
* added `lint` `test` steps on travis to be sure, that published package does not contain stupid errors
* added filter _publish only on node **4**_
* added coverage reporting (to https://codecov.io)

fix #6

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -305,26 +239,12 @@ Freshdesk.prototype = Object.create({
 	 * @param  {integer}                            id      Company ID
 	 * @param  {Freshdesk.requestCallback}          cb      Callback function {@link Freshdesk.requestCallback}
 	 */
-	deleteCompany(id, data, cb) {
```
Comment (maxkoryukov) [inline lib/client.js]: thanks, ESlint;)

**Comment by maxkoryukov** [comment]
Additionaly, I could say, that I have a small app, which uses this library. So, after this PR my application still work well!

It is a forced replacement for rich testing (currently, there is just one test file :cry: )

**Comment by arjunkomath** [comment]
Since its a private API, we won't be able to write all test cases :(

**Comment by arjunkomath** [comment]
@maxkoryukov Added gulp for auto bumping the version. Command `npm run release` will bump package version, tag and push. It will be tested by CI and automatically deployed to npm.

**Pull request merged**